numba/unicode.py

```diff
@@ -440,6 +453,80 @@ def endswith_impl(a, b):
         return endswith_impl
 
 
+@overload_method(types.UnicodeType, 'split')
+def unicode_split(a, sep, maxsplit=-1):
+    if isinstance(sep, types.UnicodeType):
+        def split_impl(a, sep, maxsplit=-1):
+            a_len = len(a)
+            sep_len = len(sep)
+
+            if sep_len == 0:
+                raise ValueError('empty separator')
+
+            parts = []
+            last = 0
+            idx = 0
+            split_count = 0
+
+            while idx < a_len and (maxsplit == -1 or split_count < maxsplit):
+                if _cmp_region(a, idx, sep, 0, sep_len) == 0:
+                    parts.append(a[last:idx])
+                    idx += sep_len
+                    last = idx
+                    split_count += 1
+                else:
+                    idx += 1
+
+            if last <= a_len:
+                parts.append(a[last:])
+
+            return parts
+        return split_impl
+
+
+@njit
+def join_list(sep, parts):
+    parts_len = len(parts)
+    if parts_len == 0:
+        return ''
+
+    # Precompute size and char_width of result
+    sep_len = len(sep)
+    length = (parts_len - 1) * sep_len
+    kind = sep._kind
+    for p in parts:
+        length += len(p)
+        kind = _pick_kind(kind, p._kind)
```

Comment:
tiny performance hack, once kind is max kind, stop picking kinds?

Reply:
If I am super paranoid, this would allow strings of the unsupported WCHAR variety to slip through, rather than raise an exception.  However, I think it should be impossible at this point in the code, so perhaps the `_pick_kind` logic has moved from defensive to paranoid.

Reply:
Think this can be left until it is of concern, the performance difference probably won't be large.